ext/src/ruby_api/externals.rs

```diff
@@ -1,13 +1,98 @@
 use super::{
-    convert::WrapWasmtimeType, func::Func, global::Global, memory::Memory, root,
-    store::StoreContextValue, table::Table,
+    convert::{WrapWasmtimeExternType, WrapWasmtimeType},
+    func::{Func, FuncType},
+    global::{Global, GlobalType},
+    memory::{Memory, MemoryType},
+    root,
+    store::StoreContextValue,
+    table::{Table, TableType},
 };
 use crate::{conversion_err, not_implemented};
 use magnus::{
     class, gc::Marker, method, prelude::*, rb_sys::AsRawValue, typed_data::Obj, DataTypeFunctions,
     Error, Module, RClass, Ruby, TypedData, Value,
 };
 
+#[derive(TypedData)]
+#[magnus(
+    class = "Wasmtime::ExternType",
+    size,
+    mark,
+    free_immediately,
+    unsafe_generics
+)]
+pub enum ExternType {
+    Func(Obj<FuncType>),
+    Global(Obj<GlobalType>),
+    Memory(Obj<MemoryType>),
+    Table(Obj<TableType>),
+}
+
+impl DataTypeFunctions for ExternType {
+    fn mark(&self, marker: &Marker) {
+        match self {
+            ExternType::Func(f) => marker.mark(*f),
+            ExternType::Global(g) => marker.mark(*g),
+            ExternType::Memory(m) => marker.mark(*m),
+            ExternType::Table(t) => marker.mark(*t),
+        }
+    }
+}
+unsafe impl Send for ExternType {}
+
+impl ExternType {
+    /// @yard
+    /// Returns the exported function's FuncType or raises a `{ConversionError}` when the export is not a
+    /// function.
+    /// @return [Func] The exported function.
```

Comment:
I believe here it should be

```
@return [FuncType]
```

In the signatures below as well?

Reply:
Good catch! Fixed.